static/js/lib/api.js

```diff
@@ -17,6 +18,15 @@ export function getChannel(channelName: string): Promise<Channel> {
   return fetchJSONWithCSRF(`/api/v0/channels/${channelName}/`);
 }
 
+export function createChannel(channel: Channel): Promise<Channel> {
+  return fetchJSONWithCSRF(`/api/v0/channels/`, {
+    method: POST,
+    body: JSON.stringify({
+      ...channel
```

Comment:
Yeah, that's less than useful, it also isn't guarding against the other channel properties which could cause a 4xx.